### ISSUE TITLE
Two `BasicsTest` fails with pg 1.5.0

### ISSUE THREAD

**Issue by yahonda** [rails/rails#48049]
According to https://github.com/rails/rails/pull/48048 Two of Active Record tests fails with pg gem 1.5.0 released on Apr 24, 2023 https://github.com/ged/ruby-pg/blob/master/History.md#v150-2023-04-24-lars-kanis-larsgreiz-reinsdorfde.

### Steps to reproduce

```ruby
git clone https://github.com/rails/rails
cd rails 
rm Gemfile.lock
bundle install
cd activerecord
ARCONN=postgresql bin/test test/cases/base_test.rb -n /test_preserving_time_objects/
```

### Expected behavior
It should pass.

### Actual behavior
It fails as follows.

```ruby
$ bundle info pg
  * pg (1.5.0)
	Summary: Pg is the Ruby interface to the PostgreSQL RDBMS
	Homepage: https://github.com/ged/ruby-pg
	Documentation: http://deveiate.org/code/pg
	Source Code: https://github.com/ged/ruby-pg
	Changelog: https://github.com/ged/ruby-pg/blob/master/History.md
	Path: /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/pg-1.5.0
	Reverse Dependencies:
		queue_classic (4.0.0) depends on pg (>= 1.1, < 2.0)
$ ARCONN=postgresql bin/test test/cases/base_test.rb -n /test_preserving_time_objects/
Using postgresql
PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
Run options: -n /test_preserving_time_objects/ --seed 27151

# Running:

PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
F

Failure:
BasicsTest#test_preserving_time_objects_with_time_with_zone_conversion_to_default_timezone_local [/home/yahonda/src/github.com/rails/rails/activerecord/test/cases/base_test.rb:324]:
--- expected
+++ actual
@@ -1 +1 @@
-Sat, 01 Jan 2000 00:00:00.000000000 CST -06:00
+2000-01-01 01:00:00 UTC



bin/test test/cases/base_test.rb:317

PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
...PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
F

Failure:
BasicsTest#test_preserving_time_objects_with_utc_time_conversion_to_default_timezone_local [/home/yahonda/src/github.com/rails/rails/activerecord/test/cases/base_test.rb:310]:
Expected: 2000-01-01 00:00:00 UTC
  Actual: 1999-12-31 19:00:00 UTC


bin/test test/cases/base_test.rb:304



Finished in 0.169755s, 29.4541 runs/s, 82.4716 assertions/s.
5 runs, 14 assertions, 2 failures, 0 errors, 0 skips
$
```

### System configuration
**Rails version**: main branch

**Ruby version**: ruby 3.2.2 (2023-03-30 revision e51014f9c0) [x86_64-linux]

**Comment by yahonda** [comment]
This warning will be addressed via https://github.com/rails/rails/pull/48048
```
PG::Coder.new(hash) is deprecated. Please use keyword arguments instead! Called from /home/yahonda/src/github.com/rails/rails/activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb:1099:in `new'
```

**Comment by yahonda** [comment]
According to `git bisect` these two failures are triggered since https://github.com/ged/ruby-pg/commit/fd8bdbbcd0f270e1368a4a56bf6781494e96f1d7

**Comment by larskanis** [comment]
The regression is fixed in pg-1.5.1, which I just released.

The warning about keyword arguments is still present and should be fixed in rails like https://github.com/rails/rails/pull/48048 does.

Sorry for the inconvenience!

**Comment by yahonda** [comment]
Confirmed it is fixed by https://github.com/ged/ruby-pg/pull/525
